pkg/renderer/sgml/aliases.go

```diff
@@ -0,0 +1,23 @@
+package sgml
+
+import (
+	html "html/template"
+	text "text/template"
+
+	"github.com/bytesparadise/libasciidoc/pkg/renderer"
+	"github.com/bytesparadise/libasciidoc/pkg/types"
+)
+
+// These type aliases provide local names for names in other packages,
+// thereby helping minimize collisions based on conflicting package
+// names.  It also reduces the imports we have to use everywhere else.
+
+type Context = renderer.Context
```

Comment:
> Using aliases means that I get to have less namespace pollution, which I think was part of why this was important to me. In particular, the renderer.Context was really annoying to me, because I couldn't use renderer as a symbol name anywhere else.

I see what you mean. I wished the alias could allow for a `renderer` type (not just a var): I tried renaming the `sgmlRenderer` type to `renderer` but it did not work :(

One reason why I don't like such aliases is that VSCode cannot "traverse" the alias and display the struct in the popup dialog when I hover the type

Eg:
![image](https://user-images.githubusercontent.com/178502/84354015-78fae300-abc0-11ea-88cf-56803ca56f90.png)

vs 

![image](https://user-images.githubusercontent.com/178502/84354103-962fb180-abc0-11ea-88ac-3e17ff0c777b.png)

> Basically the type alias like type Context = renderer.Context makes this symbol name a first class entity in my package (and exported), so there are no name collisions and lets me give a nice clean interface boundary to my package consumers.

sure, but it's a bit misleading IMO for the external developers, and anyways, the `pkg/renderer` is also public, so things should be clear IMO.

Reply:
The alias traversal not working in VSCode is unfortunate.  GoLand handles them beautifully.  I switch to GoLand from VSC a while ago, and I couldn't be happier with it.

Reply:
> This kind of type aliasing is relatively new (go 1.11 or something like that).
> 
Yes, but IIRC, it was aimed at helping developers make a transition from one type to another. I wasn't aware of this use case.

> I've found it really helps to reduce the interface boundary between things. And narrow interfaces are something that I consider always a good design goal.
> 
> The other thing, is that having local aliases means that the code is far more readable -- at least to folks who aren't used to reading a lot of this,package.from.that.thing.but.imported.here.variable style of coding (Java).
> 
well, in go there's a single level of package, so I'm not sure what's the problem here?


> Meaning, to my mind, open(doc Document) is easier to read than open(doc types.Documents)
> 
to be honest I'm totally used to `open(doc types.Documents)`

> It's a small thing taken one at a time, but over time it gets to be ... annoying to type, and it disrupts the flow of the code.
> 

On the contrary, I believe it adds some mental load because I would always to remember that `Document` is an `Alias` for `types.Document` and if I want to navigate to the struct definition, I will have to go through the alias, first.